### PR TITLE
Added custom logger with levels and encoders

### PR DESCRIPTION
**Pull request by igor-karpukhin**
### All Submissions:
This PR adds custom zap logger with configurable log level and log encoder
Resolves: https://github.com/mongodb/mongodb-atlas-kubernetes/issues/470
* [x] Have you signed our [CLA](https://www.mongodb.com/legal/contributor-agreement)?

### CODE REVIEW

File: cmd/manager/main.go [hunk 1]
```diff
@@ -192,7 +202,8 @@ func parseConfiguration(log *zap.SugaredLogger) Config {
 	flag.BoolVar(&config.EnableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
-
+	flag.StringVar(&config.LogLevel, "log-level", "info", "Log level. Available values: debug | info | warn | error | dpanic | panic | fatal")
```
Comment (jianrongzhang89) [inline cmd/manager/main.go:205]: log-level and log-encoder should be added to the pod configuration manager.yaml as well.

Comment (igor-karpukhin) [reply cmd/manager/main.go:205]: Fixed

File: cmd/manager/main.go [hunk 2]
```diff
@@ -203,7 +214,7 @@ func parseConfiguration(log *zap.SugaredLogger) Config {
 	config.WatchedNamespaces = make(map[string]bool)
 	for _, namespace := range strings.Split(watchedNamespace, ",") {
 		namespace = strings.TrimSpace(namespace)
-		log.Infof("The Operator is watching the namespace %s", namespace)
+		fmt.Printf("The Operator is watching the namespace %s\r\n", namespace)
```
Comment (jianrongzhang89) [inline cmd/manager/main.go:217]: Can this be done through the logger instead of printf? If so it will be more consistent.

**Review by fabritsius — APPROVED** [review]
LGTM 👍

File: cmd/manager/main.go (hunk 2 above)
Comment (igor-karpukhin) [reply cmd/manager/main.go]: Remove this one. Added config printing when application starts.

File: cmd/manager/main.go (hunk 1 above)
Comment (jianrongzhang89) [reply cmd/manager/main.go:207]: Should files under deploy and bundle sub-directories be updated also?

**Review by leo-ri — APPROVED** [review]

**Pull request merged**